### PR TITLE
Added: `__version__` attribute

### LINKED ISSUE #197: make os independent development environment
Currently - Mac users cannot run any of the `make` commands for the repo - which forces repo maintainers to reformat code written by Mac users. 

This applies to both `ipyvizzu` and `ipyvizzu-story`

### LINKED ISSUE #228: use Poetry with pyproject.toml
Instead of setup.py create a pyproject.toml file.

- [x] add a pyproject.toml draft file
- [x] separate extras dependencies (fugue) in the pyproject.toml file
- [ ] modify github workflows
- [ ] use task runner (for example poe) instead of makefile #197

### LINKED ISSUE #229: use tox and run tests with different python versions
Use tox, configure it with different python versions:

  - [ ] 3.7
  - [ ] 3.8
  - [ ] 3.9
  - [ ] 3.10
  - [ ] 3.11

reconfigure ci workflow to run steps with different python versions.

### PR DESCRIPTION
**Pull request by veghdev**
- add __version__ attribute
- add pyproyect.toml instead of setup.py, resolve #228 
- use pdm for managing, locking development dependencies instead of pip-compile
- use pdm for package building instead of build
- use pdm.scripts instead of makefile, resolve #197 
- use tox for testing with multiple python version, resolve #229 
- keep pylint missing-docstring only in the src
- refactor the tools folder

Most of the modifications:

fix formatting with the newest version of prettier, black, mdformat
keep docstring only in the src
rename data_frame variables to df

Important modifications:

src/ipyvizzu/init.py
src/ipyvizzu/version.py
pyproject.toml (instead of setup.py)
pyproject.toml scripts (instead of Makefile)
CONTRIBUTING.md
refactor in the tools/ folder
refactor in the .github/workflows folder

### CODE REVIEW

**Comment by veghdev** [comment]
Most of the modifications:
- fix formatting with the newest version of prettier
- keep docstring only in the src
- rename data_frame variables to df

Important modification:
- src/ipyvizzu/__init__.py
- src/ipyvizzu/__version__.py
- pyproject.toml (instead of setup.py)
- pyproject.toml scripts (instead of Makefile)
- CONTRIBUTING.md

**Review by schaumb — DISMISSED** [review]
tldr ship it

**Review by schaumb — APPROVED** [review]
